invenio_rdm_records/services/components/custom_fields.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020-2021 CERN.
+# Copyright (C) 2020 Northwestern University.
+# Copyright (C) 2021 TU Wien.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""RDM service component for metadata."""
+
+from copy import copy
+
+from invenio_drafts_resources.services.records.components import ServiceComponent
+
+
+class CustomFieldsComponent(ServiceComponent):
+    """Service component for custom fields."""
+
+    def create(self, identity, data=None, record=None, **kwargs):
+        """Inject parsed metadata to the record."""
+        record["custom"] = data.get("custom", {})
+
+    def update_draft(self, identity, data=None, record=None, **kwargs):
+        """Inject parsed metadata to the record."""
+        record["custom"] = data.get("custom", {})
+
+    def publish(self, identity, draft=None, record=None, **kwargs):
+        """Update draft metadata."""
+        record["custom"] = draft.get("custom", {})
+
+    def edit(self, identity, draft=None, record=None, **kwargs):
+        """Update draft metadata."""
+        draft["custom"] = record.get("custom", {})
+
+    def new_version(self, identity, draft=None, record=None, **kwargs):
```

Comment:
is this needed? shouldn't they be passed forward? why do they (and metadata) get cleaned?

Reply:
https://github.com/inveniosoftware/invenio-rdm-records/issues/1002